src/shared/dlt_common.c

```diff
@@ -2431,7 +2431,10 @@ void dlt_buffer_write_block(DltBuffer *buf, int *write, const unsigned char *dat
                 *write += (int) size;
             }
             else {
-                if(buf->size > (unsigned int) (*write)) {
+                /* when (*write) = buf->size, write only the second block
+                * and update write position correspondingly.
+                */
+                if((unsigned int) (*write) <= buf->size) {
```

Comment:
Hi @MilanRusko , 
I have updated this commit for correcting write/read position. Can you take a look on it and give feedback? 

Thank you in advance.